core/src/com/unciv/logic/automation/UnitAutomation.kt

```diff
@@ -23,6 +20,38 @@ class UnitAutomation {
     companion object {
         const val CLOSE_ENEMY_TILES_AWAY_LIMIT = 5
         const val CLOSE_ENEMY_TURNS_AWAY_LIMIT = 3f
+
+        internal fun tryExplore(unit: MapUnit, unitDistanceToTiles: PathsToTilesWithinTurn): Boolean {
+            if (tryGoToRuin(unit, unitDistanceToTiles) && unit.currentMovement == 0f) return true
+
+            for (tile in unit.currentTile.getTilesInDistance(10))
+                if (unit.movement.canMoveTo(tile) && tile.position !in unit.civInfo.exploredTiles
+                        && unit.movement.canReach(tile)
+                        && (tile.getOwner() == null || !tile.getOwner()!!.isCityState())) {
+                    unit.movement.headTowards(tile)
+                    return true
+                }
+            return false
+        }
+
+        private fun tryGoToRuin(unit: MapUnit, unitDistanceToTiles: PathsToTilesWithinTurn): Boolean {
+            if (!unit.civInfo.isMajorCiv()) return false // barbs don't have anything to do in ruins
+            val tileWithRuin = unitDistanceToTiles.keys
+                    .firstOrNull { it.improvement == Constants.ancientRuins && unit.movement.canMoveTo(it) }
+                    ?: return false
```

Comment:
ok will do. Though this was suggested by IDE and this seems not too confusing.  the elvis operator means `if null then`, so that reads `if null, then return false`.  

Essentially, no use of elvis operators (`?:`)?

Reply:
Personally I think it's best to avoid them as much as possible but they do have their uses
Elvis operators are fine if they assign a value (eg a = b ?: c) but when they give to do an action, then you're mixing an assignment statement and an action statement, and that's less clear